Hi @stuntguy3000 

Unfortunately, this PR does not compile:

``` log
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /home/travis/build/DSH105/Nexus/src/main/java/com/dsh105/nexus/command/module/github/GithubIssueCommand.java:[27,7] error: class GitHubIssueCommand is public, should be declared in a file named GitHubIssueCommand.java
[INFO] 1 error
[INFO] -------------------------------------------------------------
```

I believe the 'h' in the filename should be capitalised. Did you test these changes?

Aside from those issues, I do like this - will be necessary for expansion in the future :smile:

--- a/src/main/java/com/dsh105/nexus/Nexus.java
+++ b/src/main/java/com/dsh105/nexus/Nexus.java
@@ -18,7 +18,7 @@
 package com.dsh105.nexus;
 
 import com.dsh105.nexus.command.CommandManager;
-import com.dsh105.nexus.command.module.RemindCommand;
+import com.dsh105.nexus.command.module.general.RemindCommand;
 import com.dsh105.nexus.config.OptionsConfig;
 import com.dsh105.nexus.hook.github.GitHub;
 import com.dsh105.nexus.hook.jenkins.Jenkins;

--- a/src/main/java/com/dsh105/nexus/command/module/admin/ExitCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/admin/ExitCommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.admin;
 
 import com.dsh105.nexus.Nexus;
 import com.dsh105.nexus.command.Command;

--- a/src/main/java/com/dsh105/nexus/command/module/general/ChannelStatsCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/general/ChannelStatsCommand.java
@@ -1,4 +1,4 @@
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.general;
 
 import com.dsh105.nexus.command.Command;
 import com.dsh105.nexus.command.CommandModule;

--- a/src/main/java/com/dsh105/nexus/command/module/general/CurrencyCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/general/CurrencyCommand.java
@@ -1,4 +1,4 @@
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.general;
 
 import com.dsh105.nexus.command.Command;
 import com.dsh105.nexus.command.CommandModule;

--- a/src/main/java/com/dsh105/nexus/command/module/general/HelpCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/general/HelpCommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.general;
 
 import com.dsh105.nexus.Nexus;
 import com.dsh105.nexus.command.Command;

--- a/src/main/java/com/dsh105/nexus/command/module/general/IpInfoCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/general/IpInfoCommand.java
@@ -1,4 +1,4 @@
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.general;
 
 import com.dsh105.nexus.command.Command;
 import com.dsh105.nexus.command.CommandModule;
@@ -40,6 +40,7 @@ public class IpInfoCommand extends CommandModule {
             HttpResponse<JsonNode> resp = Unirest.get(url).asJson();
 
             if (resp.getCode() != 200) {
+                event.errorWithPing("Unknown error occurred while executing command! Try again later...");
                 return false;
             }
 

--- a/src/main/java/com/dsh105/nexus/command/module/general/RemindCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/general/RemindCommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.general;
 
 import com.dsh105.nexus.Nexus;
 import com.dsh105.nexus.command.Command;

--- a/src/main/java/com/dsh105/nexus/command/module/github/GitHubIssueCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/github/GitHubIssueCommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.github;
 
 import com.dsh105.nexus.Nexus;
 import com.dsh105.nexus.command.Command;
@@ -23,8 +23,8 @@ import com.dsh105.nexus.command.CommandModule;
 import com.dsh105.nexus.command.CommandPerformEvent;
 import com.dsh105.nexus.util.StringUtil;
 
-@Command(command = "issue", needsChannel = false, help = "Retrieve issue information for a GitHub repository", extendedHelp = "This command is simply an alias of {p}{b}repo <name> issue <number>{/b}")
-public class IssueCommand extends CommandModule {
+@Command(command = "githubissue", aliases = {"ghi", "issue"}, needsChannel = false, help = "Retrieve issue information for a GitHub repository", extendedHelp = "This command is simply an alias of {p}{b}repo <name> issue <number>{/b}")
+public class GitHubIssueCommand extends CommandModule {
     @Override
     public boolean onCommand(CommandPerformEvent event) {
         if (event.getArgs().length == 2 || event.getArgs().length == 3) {

--- a/src/main/java/com/dsh105/nexus/command/module/github/GitHubKeyCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/github/GitHubKeyCommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.github;
 
 import com.dsh105.nexus.Nexus;
 import com.dsh105.nexus.command.Command;
@@ -34,9 +34,9 @@ import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.HashMap;
 
-@Command(command = "ghkey", aliases = "ghk", needsChannel = false, help = "Authenticate with GitHub through Nexus to allow the use of various GitHub commands requiring an API key.",
+@Command(command = "githubkey", aliases = {"ghk", "ghkey"}, needsChannel = false, help = "Authenticate with GitHub through Nexus to allow the use of various GitHub commands requiring an API key.",
         extendedHelp = {"{b}{p}{c}{/b} - Provides instructions on how to set this up."})
-public class GhKeyCommand extends CommandModule {
+public class GitHubKeyCommand extends CommandModule {
 
     private static String AUTHORISE = "https://github.com/login/oauth/authorize?client_id={client_id}&scope={scope}&state={state}";
     private static String ACCESS_TOKEN = "https://github.com/login/oauth/access_token?client_id={client_id}&client_secret={client_secret}&code={code}";

--- a/src/main/java/com/dsh105/nexus/command/module/github/GitHubRepositoryCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/github/GitHubRepositoryCommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.github;
 
 import com.dsh105.nexus.Nexus;
 import com.dsh105.nexus.command.Command;
@@ -30,7 +30,7 @@ import org.pircbotx.Colors;
 import java.util.ArrayList;
 import java.util.regex.Pattern;
 
-@Command(command = "repo", needsChannel = false, help = "Management of GitHub repositories.",
+@Command(command = "githubrepository", aliases = {"repo", "repository", "ghrepo"}, needsChannel = false, help = "Management of GitHub repositories.",
         extendedHelp = {
                 "The repo command contains various commands to manage GitHub repositories.",
                 "{b}{p}{c} <name>{/b} - retrieves repository information for the given repo. Uses the sender's nick as the GitHub login",
@@ -41,7 +41,7 @@ import java.util.regex.Pattern;
                 "{b}{p}{c} <...> set <option> <args>{/b} - sets the value of the given event option for a repo.",
                 "{b}{p}{c} <..> get <option>{/b} - retrieves information on the given event option.",
                 "Valid options are: irc"})
-public class RepoCommand extends CommandModule {
+public class GitHubRepositoryCommand extends CommandModule {
 
     @Override
     public boolean onCommand(CommandPerformEvent event) {

--- a/src/main/java/com/dsh105/nexus/command/module/github/GitHubUserCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/github/GitHubUserCommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.github;
 
 import com.dsh105.nexus.command.Command;
 import com.dsh105.nexus.command.CommandModule;
@@ -25,19 +25,11 @@ import com.dsh105.nexus.hook.github.GitHub;
 import com.dsh105.nexus.hook.github.GitHubUser;
 import com.dsh105.nexus.util.StringUtil;
 import com.dsh105.nexus.util.shorten.URLShortener;
-import com.mashape.unirest.http.HttpResponse;
-import com.mashape.unirest.http.JsonNode;
-import com.mashape.unirest.http.Unirest;
-import com.mashape.unirest.http.exceptions.UnirestException;
-import org.json.JSONObject;
 import org.pircbotx.Colors;
 
-import java.io.UnsupportedEncodingException;
-import java.net.URLEncoder;
-
-@Command(command = "ghuser", aliases = {"ghu"}, needsChannel = false, help = "Retrieves information on a GitHub user",
+@Command(command = "githubuser", aliases = {"ghu", "ghuser"}, needsChannel = false, help = "Retrieves information on a GitHub user",
         extendedHelp = {"{b}{p}{c}{/b} <user_name> - Provides information on a GitHub user."})
-public class GhUserCommand extends CommandModule {
+public class GitHubUserCommand extends CommandModule {
 
     @Override
     public boolean onCommand(CommandPerformEvent event) {

--- a/src/main/java/com/dsh105/nexus/command/module/jenkins/BuildCommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/jenkins/BuildCommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.jenkins;
 
 import com.dsh105.nexus.Nexus;
 import com.dsh105.nexus.command.Command;

--- a/src/main/java/com/dsh105/nexus/command/module/jenkins/CICommand.java
+++ b/src/main/java/com/dsh105/nexus/command/module/jenkins/CICommand.java
@@ -15,7 +15,7 @@
  * along with Nexus.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-package com.dsh105.nexus.command.module;
+package com.dsh105.nexus.command.module.jenkins;
 
 import com.dsh105.nexus.Nexus;
 import com.dsh105.nexus.command.Command;

